packages/block-editor/src/store/selectors.js

```diff
@@ -2542,7 +2542,18 @@ function getReusableBlocks( state ) {
  * @return {boolean} Is navigation mode enabled.
  */
 export function isNavigationMode( state ) {
-	return state.isNavigationMode;
+	return state.editorMode === 'navigation';
+}
+
+/**
+ * Returns the current editor mode.
+ *
+ * @param {Object} state Editor state.
+ *
+ * @return {string} the editor mode.
+ */
+export function __unstableGetEditorMode( state ) {
```

Comment:
I wonder, should I switch to this approach in https://github.com/WordPress/gutenberg/pull/41740 as another mode?